model_analyzer/record/metrics_manager.py

```diff
@@ -166,21 +166,22 @@ def execute_run_config(self, run_config):
         measurement to the result manager
```

Comment:
This function does a lot. Can we refactor into some smaller functions so it is easier to read? (At least do the code you touched)

Reply:
refactored into smaller functions 👍